src/execution/compiler/operator/output_translator.cpp

```diff
@@ -39,11 +45,19 @@ void OutputTranslator::PerformPipelineWork(terrier::execution::compiler::WorkCon
     ast::Expr *rhs = child_translator->GetOutput(context, attr_idx);
     function->Append(GetCodeGen()->Assign(lhs, rhs));
   }
+
+  CounterAdd(function, num_output_, 1);
 }
 
 void OutputTranslator::FinishPipelineWork(const Pipeline &pipeline, FunctionBuilder *function) const {
   auto exec_ctx = GetExecutionContext();
   function->Append(GetCodeGen()->CallBuiltin(ast::Builtin::ResultBufferFinalize, {exec_ctx}));
+
+  FeatureRecord(function, brain::ExecutionOperatingUnitType::OUTPUT,
+                brain::ExecutionOperatingUnitFeatureAttribute::NUM_ROWS, pipeline, CounterVal(num_output_));
+  FeatureRecord(function, brain::ExecutionOperatingUnitType::OUTPUT,
```

Comment:
A minor thing: we should remove setting the cardinality for `OUTPUT` since you don't really know which consumer it is and it would be set outside.